src/components/hooks/__tests__/use-dynamic-presets.test.ts

```
@@ -0,0 +1,243 @@
+    addHours: vi.fn(),
+  }
+})
+
+describe('useDynamicPresets', () => {
+  const mockDb = {
+    getScheduledPosts: vi.fn(),
+  }
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+    ;(db as any).mockReturnValue(mockDb)
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)